pkg/secrets-store/nodeserver_test.go

```diff
@@ -84,14 +96,154 @@ func TestNodePublishVolume(t *testing.T) {
 			},
 			expectedErr: true,
 		},
+		{
+			name: "secret provider class in pod namespace not found",
+			nodePublishVolReq: csi.NodePublishVolumeRequest{
+				VolumeCapability: &csi.VolumeCapability{},
+				VolumeId:         "testvolid1",
+				TargetPath:       getTestTargetPath(t),
+				VolumeContext:    map[string]string{"secretProviderClass": "provider1", csipodname: "pod1", csipodnamespace: "default"},
+			},
+			initObjects: []runtime.Object{
+				&v1alpha1.SecretProviderClass{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "provider1",
+						Namespace: "testns",
+					},
+				},
+			},
+			expectedErr: true,
+		},
+		{
+			name: "provider not set in secret provider class",
+			nodePublishVolReq: csi.NodePublishVolumeRequest{
+				VolumeCapability: &csi.VolumeCapability{},
+				VolumeId:         "testvolid1",
+				TargetPath:       getTestTargetPath(t),
+				VolumeContext:    map[string]string{"secretProviderClass": "provider1", csipodname: "pod1", csipodnamespace: "default"},
+			},
+			initObjects: []runtime.Object{
+				&v1alpha1.SecretProviderClass{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "provider1",
+						Namespace: "default",
+					},
+				},
+			},
+			expectedErr: true,
+		},
+		{
+			name: "parameters not set in secret provider class",
+			nodePublishVolReq: csi.NodePublishVolumeRequest{
+				VolumeCapability: &csi.VolumeCapability{},
+				VolumeId:         "testvolid1",
+				TargetPath:       getTestTargetPath(t),
+				VolumeContext:    map[string]string{"secretProviderClass": "provider1", csipodname: "pod1", csipodnamespace: "default"},
+			},
+			initObjects: []runtime.Object{
+				&v1alpha1.SecretProviderClass{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "provider1",
+						Namespace: "default",
+					},
+					Spec: v1alpha1.SecretProviderClassSpec{
+						Provider: "provider1",
+					},
+				},
+			},
+			expectedErr: true,
+		},
+		{
+			name: "read only is not set to true",
+			nodePublishVolReq: csi.NodePublishVolumeRequest{
+				VolumeCapability: &csi.VolumeCapability{},
+				VolumeId:         "testvolid1",
+				TargetPath:       getTestTargetPath(t),
+				VolumeContext:    map[string]string{"secretProviderClass": "provider1", csipodname: "pod1", csipodnamespace: "default"},
+			},
+			initObjects: []runtime.Object{
+				&v1alpha1.SecretProviderClass{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "provider1",
+						Namespace: "default",
+					},
+					Spec: v1alpha1.SecretProviderClassSpec{
+						Provider:   "provider1",
+						Parameters: map[string]string{"parameter1": "value1"},
+					},
+				},
+			},
+			expectedErr: true,
+		},
+		{
+			name: "failed to invoke provider, unmounted to force retry",
```

Comment:
where do we test retry?

Reply:
Added another loop just to test retry for all the failed calls